src/main/resources/avro/common.avdl

```diff
@@ -18,33 +337,112 @@ enum Strand {
 }
 
 /**
-An abstraction for referring to a genomic position, in relation to some
-already known reference. For now, represents a genomic position as a reference
-name, a base number on that reference (0-based), and a flag to say if it's the
-forward or reverse strand that we're talking about.
+A `Position` is a side of a base pair in some already known sequence. A
+`Position` is represented by a sequence name or ID, a base number on that
+sequence (0-based), and a `Strand` to indicate the left or right side.
+
+For example, given the sequence "GTGG", the `Position` on that sequence at
+offset 1 in the forward orientation would be the left side of the T/A base pair.
+The base at this `Position` is "T". Alternately, for offset 1 in the reverse
+orientation, the `Position` would be the right side of the T/A base pair, and
+the base at the `Position` is "A".
+
+Offsets added to a `Position` are interpreted as reading along its strand;
+adding to a reverse strand position actually subtracts from its `position`
+member.
 */
-// TODO: Add support here for universally unique base IDs.
 record Position {
   /**
-  The name of the reference (or, more technically, the scaffold) in whatever
-  reference set is being used. Does not generally include a "chr" prefix, so for
-  example "X" would be used for the X chromosome.
+  The name of the reference sequence in whatever reference set is being used.
+  Does not generally include a "chr" prefix, so for example "X" would be used
+  for the X chromosome.
+
+  If `sequenceId` is null, this must not be null.
+  */
+  union { null, string } referenceName = null;
+
+  /**
+  The ID of the sequence on which the `Position` is located. This may be a
+  `Reference` sequence, or a novel piece of sequence associated with a
+  `VariantSet`.
+
+  If `referenceName` is null, this must not be null.
   */
-  string referenceName;
+  union { null, string } sequenceId = null;
```

Comment:
It's a string because the other IDs in the API are strings. Implementations could use integers, UUIDs, accessions, or whatever else they want.